### PR TITLE
Focus selected block in editor canvas when clicking edit button on zoom out mode toolbar

### PR DESCRIPTION
**Pull request by jeryj**
## What?

Fixes a focus loss introduced in https://github.com/WordPress/gutenberg/pull/64571

## Why?

Focus loss = bad a11y

## How?

Send focus to the contentRef, which lets [writing flow handle sending the focus to the selected block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/writing-flow/use-tab-nav.js#L52-L56).

### Testing Instructions for Keyboard

- Enter zoom out mode via the pattern inserter
- Select a new top level group block
- Select the edit button
- Zoom out mode should exit
- The selected block should have focus. Test this by pressing Escape to enter navigation mode and see if your group block is selected

### CODE REVIEW

**Review by getdave — APPROVED** [review]
Thanks for fixing this. Testing seems to indicate the section I clicked "Edit" on is focused.
